### PR TITLE
Add code of conduct

### PR DESCRIPTION
**Pull request by sferik**
See [discussion on the mailing list](https://groups.google.com/d/msg/crystal-lang/gq-t15bpXCg/houoDpz1DgAJ).

### CODE REVIEW

**Comment by jalyna** [comment]
Thank you so much for setting this up :+1: I have just one two notes (maybe they were already answered then I'm sorry)
- Wouldn't it be more readable to have a `.md` file (especially for the links)?
- Could we maybe add an email address for not public issues? Right now it says to contact the main contributors (is there even an official list of them?).

Anyway: thanks.

**Comment by Exilor** [comment]
From http://contributor-covenant.org
```
Insensitive language, thoughtless use of pronouns, projects with sexualized or culturally inappropriate names, and the pervasive cult of meritocracy make contributing to open source a daunting prospect for many people.
```
That's some thin skin. I'm gay and I'm far more worried about code I submit being buggy than I am of being called a filthy sodomite by some random nobody. Which I doubt would happen anyway, code of conduct or not.

**Comment by strcmp** [comment]
@Exilor yes I think I agree with you, I'm not sure a code of conduct would prevent anyone from engaging in behaviour like that and furthermore I doubt many people take the time to read it but some undoubtedly do (I guess it's like a LICENSE.txt file, I rarely read those). however for whatever reason, CoC seem to make a certain group of people more comfortable, yet I doubt it has prevented any kind of the behaviour it disallows or even discouraged it (in fact, the opposite could be true). It seems more like a political statement than anything useful, but I've got use to them and PRs like this by now.

**Comment by sferik** [comment]
> Wouldn't it be more readable to have a .md file (especially for the links)?

I’ve converted the file to Markdown. The links are now clickable. Thanks for the suggestion.

> Could we maybe add an email address for not public issues? Right now it says to contact the main contributors (is there even an official list of them?).

A list of contributors can be found [here](https://github.com/manastech/crystal/graphs/contributors). The main contributors to this project are @asterite and @waj, who both list their email address on their public GitHub profiles.

**Comment by porras** [comment]
Hi @Exilor,

You don't care about strangers insulting you on the internet and that's fine. But I'm sure you understand that some other people *do* care, and that that's also fine (it doesn't happen often to me, but I definitely care and don't like it). I find it nice from the projects to provide them with the signal that that won't be tolerated.

As of the concern exposed by @strcmp that a CoC wouldn't stop that from happening, I tend to agree with it, but I think it's not the point of a CoC. It's not the most polite thing to do, but since the discussion moved here from the mailing list, I'll quote myself:

> A CoC doesn't (unfortunately) avoid harassment because it's not magic. It signals to current and potential contributors that there will be a specific way to respond to it when/if it happens. The mere absence of previous incidents doesn't signal that.

(I edited minor grammar errors I made at my first try because English :wink:)

TL;DR: I'm in favor of the adoption and enforcement of a CoC. I also think that people that wasn't planning to do any of the things forbidden by it should be, if not in favor because they're not very worried about the issue, at least not opposed (that's obviously just my opinion).

**Comment by sferik** [comment]
> That's some thin skin. I'm gay and I'm far more worried about code I submit being buggy than I am of being called a filthy sodomite by some random nobody. Which I doubt would happen anyway, code of conduct or not.

I hope that would not happen. I also hope we can agree that such conduct would be unacceptable. Some—not all, but some people—have expressed that they would feel more comfortable contributing with some basic ground rules in place. If that results in even one more person contributing to this project who would not have otherwise, I think this change will have been worthwhile. If not, I’m not sure there’s any harm in publicly stating that we’ll all be civil toward each other.

**Comment by sferik** [comment]
> I'm not sure a code of conduct would prevent anyone from engaging in behaviour like that and furthermore I doubt many people take the time to read it but some undoubtedly do (I guess it's like a LICENSE.txt file, I rarely read those). however for whatever reason, CoC seem to make a certain group of people more comfortable, yet I doubt it has prevented any kind of the behaviour it disallows or even discouraged it (in fact, the opposite could be true).

You could make the same argument against most laws. A law against murder does not prevent people from killing each other, but I think we can agree that such laws should be in place and that there should be clear consequences for murder, even if that does not prevent all murders.

**Comment by alex-fedorov** [comment]
:+1:

**Comment by davydovanton** [comment]
@sferik :+1:

**Comment by strcmp** [comment]
hey. I'm not -1 or +1. I'm 0 :) it has merits when it makes people feel more welcome, but i find it formal and stern otoh.

**Comment by fowlmouth** [comment]
Please mention my group specifically, single white males. We are often being oppressed these days as in these code of conducts things such as race, gender, religion are mentioned (and protected) but no mention of straightness or whiteness. Single white male is the least popular thing to be in tech, I'm constantly being reminded that I'm a "dying breed", that my days of lording over the invisible mass of multicolored/non-males is coming to an end.
So please mention the white man, inventor of civilization, space travel, etc..

**Comment by waj** [comment]
@sferik any reason not to update to Contributor Covenant 1.2.0?
